Megafont implement camp machine prefab

Megafont opened the pull request:
+ Added Machine_Base class for all machines to inherit from
+ Added IMachine interface so code can easily interface with any machine's base features
+ Added PowerRadius and MaxPowerRadius settings to the IslandHeartPowerSystem class. This determines the distance within which a machine can receive power. It starts out pretty small for now, but increases slightly with each level up of the island heart

+ Fixed bug with the new dew item identities (some didn't have their prefabs set to the new prefabs)

Machines have a RunContinously option. If enabled, it will try to run every time the specified interval has elapsed again. It just makes it possible to implement machines that have an inventory and will process items from it over time. The test machine does not use it, though it works. It will just log a message that it processed nothing each time it runs. Right now, you can put berries in the test machine, and wood dews pop out because why not? lol 

The test machine is sitting next to the island heart.

Megafont committed:
Implemented Test Machine prefab

+ Added Machine_Base class for all machines to inherit from
+ Added IMachine interface so code can easily interface with any machine's base features
+ Added PowerRadius setting to the IslandHeartPowerSystem class. This determines the distance within which a machine can receive power. It starts out pretty small for now, but increases with each level up of the island heart

+ Fixed bug with the new dew item identities (some didn't have their prefabs set to the new prefabs)

Machines have a RunContinously option. If enabled, it will try to run every time the specified interval has elapsed again. It just makes it possible to implement machines that have an inventory and will process items from it over time.

Megafont committed:
Fixed IslandHeardPowerSystem not clamping power radius to ensure it does not exceed the specified max value

wolfyman411 commented:
-Merging to develop, change it.
-The quit button in the gui doesn't exit the interaction, you have to press esc, I'd keep the esc functionality but add the functionality to the exit button.
-It didn't look like the machine actually checked if the island heart had no power (the power restoring was fast, so maybe it does)
-I was able to use any item on the machine and it was consumed (I'd add a either a whitelist system, or a gui similar to crafting)

Megafont committed:
Fixed suggested issues

+ Added progress bar that appears above machine
+ RunsContinously mode works fully on the test machine now.
+ Made minor mod to DragAndDrop script and Interactable.cs so interactables can now specify (via their new UseItemStackInteractions flag) that they take item stacks rather than single item interactions. The machines can use both modes (stack interactions are used in RunsContinously mode so its easier to load items into them in that case)
+ Fixed the UI button. It was referencing the IslandHeart prefab still, so now it calls StopInteraction() on the machine object instead.

Megafont commented:
> -Merging to develop, change it. -The quit button in the gui doesn't exit the interaction, you have to press esc, I'd keep the esc functionality but add the functionality to the exit button. -It didn't look like the machine actually checked if the island heart had no power (the power restoring was fast, so maybe it does) -I was able to use any item on the machine and it was consumed (I'd add a either a whitelist system, or a gui similar to crafting)

Ok everything should be good to go now. I also added a progress bar that appears above the machine while it's working so you can see the delay, which can be changed in the inspector.

wolfyman411 commented:
looks good
